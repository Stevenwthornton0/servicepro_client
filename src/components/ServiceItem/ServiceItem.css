.service {
    margin-bottom: 30px;
}

.service > h2 {
    font-size: 28px;
}