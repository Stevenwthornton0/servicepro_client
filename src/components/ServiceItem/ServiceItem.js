import React, { Component } from 'react';
import { displayPhone } from '../../Utils/Utils';
import './ServiceItem.css';

class ServiceItem extends Component {

    static defaultProps = {
        service: {}
    }

    render() {
        const { service } = this.props;
        let { city, state } = service;

        if (city && state) {
            city = city.replace(/^\w/, c => c.toUpperCase());
            state = state.replace(/^\w/, c => c.toUpperCase());
        }

        return (
            <div className='service'>
                <h2>{service.name}</h2>
                <p className='bold'>{city}, {state}</p>
                <p className='bold'>Phone: {displayPhone(service)}</p>
                <p className='bold'>Email: <span className='notBold'>{service.email}</span></p>
                <p>{service.about}</p>
            </div>
        )
    }
}

export default ServiceItem;