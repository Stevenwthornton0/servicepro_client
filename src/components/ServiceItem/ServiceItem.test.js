import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import ServiceItem from './ServiceItem';

Enzyme.configure({ adapter: new Adapter() })

describe('ServiceItem component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<ServiceItem />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})