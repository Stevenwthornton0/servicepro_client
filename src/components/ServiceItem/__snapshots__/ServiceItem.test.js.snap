// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceItem component renders without crashing 1`] = `
<div
  className="service"
>
  <h2 />
  <p
    className="bold"
  >
    Phone: 
  </p>
  <p
    className="bold"
  >
    Email: 
    <span
      className="notBold"
    />
  </p>
  <p />
</div>
`;
